RobotController/robot_brain.py

```diff
@@ -61,23 +65,35 @@ def _check_next_evaluation(self, force=False):
         Check if is a moment for a new evaluation and starts a new one
         :param force: forces the new evaluation to start
         """
+        if self.mating_client and self.mating_client.availability:
+            self.mating_client.server.join()
+            self.mating_client.client.join()
+            self.mating_client = None
+
         current_check = time.time()
         light_level = 1 + (self.HAL.sensor.readADC(0) / -255)
         if force or current_check > self._next_check:
             self.HAL.led.setColor(self.HAL.led._magenta)
-            logging.info("next movement values current {}, next {}".format(current_check, self._next_check))
+            logging.info("Next movement values current {}, next {}"
+                         .format(current_check, self._next_check))
             if force:
                 self.algorithm.skip_evaluation()
             else:
                 # TODO make HAL smarter in light readings
-                self.algorithm.next_evaluation(1 + (self.HAL.sensor.readADC(0) / -255))  # 255-0 to 0-1
+                self.algorithm.next_evaluation(
+                    1 + (self.HAL.sensor.readADC(0) / -255))  # 255-0 to 0-1
             self._next_check = current_check + self.TIME_CHECK_TIMEOUT
+
         if light_level < self.LIGHT_THRESHOLD:
             self.HAL.led.setColor(self.HAL.led._green)
         else:
             self.HAL.led.setColor(self.HAL.led._red)
-            if self.mating_client is None:
+            if not self.mating_client and\
+                    self.evaluations_after_mating < MATE_STEPS_THRESHOLD:
+                self.evaluations_after_mating = 0
                 self.mating_client = EvolutionaryRobot(self.robot_name)
+            else:
+                self.evaluations_after_mating += 1
```

Comment:
Wrong, this function executes every millisecond or so, not every 30 seconds. Sorry for the misleading function name...